crates/oxide/Cargo.toml

```diff
@@ -19,6 +19,7 @@ bexpand = "1.2.0"
 fast-glob = "0.4.3"
 classification-macros = { path = "../classification-macros" }
 regex = "1.11.1"
+fancy-regex = "0.14.0"
```

Comment:
I'm not sure why the `regex` library alone is not enough?

Reply:
It's because the `regex` crate doesn't support backreferences:
<img width="607" alt="image" src="https://github.com/user-attachments/assets/7381f482-bbc8-4c28-b4db-6c427e8ccbee" />


https://github.com/rust-lang/regex?tab=readme-ov-file#:~:text=This%20includes%2C%20but%20is%20not%20limited%20to%2C%20look%2Daround%20andbackreferences.%20In%20exchange%2C%20all%20regex%20searches%20in%20this%20crate%20have%20worst%20caseO(m%20*%20n)%20time%20complexity%2C%20where%20m%20is%20proportional%20to%20the%20size%20of%20the%20regexand%20n%20is%20proportional%20to%20the%20size%20of%20the%20string%20being%20searched.

But luckily `fancy-regex` does support it. But as the `regex` crate also shows is that it is technically much slower (in my testing it's negligible for our use case though)